AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -218,15 +216,18 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
         }
 
         // TODO: Wrap this in a transaction
-        if remoteModel.syncMetadata.deleted {
-            saveDeleteMutation(storageAdapter: storageAdapter, remoteModel: remoteModel)
+        if mutationType == .delete {
+            saveDeleteMutation(storageAdapter: storageAdapter, remoteModel: remoteModel, mutationType: mutationType)
```

Comment:
We know it's a delete mutation; do we need to pass the mutationType?

Reply:
the requirement is to notify the hub of `syncReceived` with the mutationType inside a MutationEvent. it's done after the two SQL operations (delete or save/update and save metadata) are done successfully, which is why it's passed down. I see now that `saveDeleteMutation` can just call `saveMetadata` with `.delete` but saveCreateOrUpdateMutation will need it. I can make this change now, and work on making the methods clearer, ie. use callbacks instead of calling each method directly